runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb

```diff
@@ -41,7 +41,7 @@ def get_long_running_operation_result(azure_response, custom_deserialization_blo
       if !AsyncOperationStatus.is_terminal_status(polling_state.status)
         task = Concurrent::TimerTask.new do
           begin
-            if !polling_state.azure_async_operation_header_link.nil?
+            if !polling_state.azure_async_operation_header_link.nil? && custom_deserialization_block.nil?
```

Comment:
@amarzavery here's a more specific question to you, could you give us some more context on the use of the operation header link versus the location header link and how we should be consuming them? Our code without this change is not using deserialization knowledge for the operation header, though this knowledge is used to deserialize the response when no operation header link exists and location header is present. I'd like to better understand if there's a reason for that, or just a bug in our code.